src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -36,20 +36,19 @@ public StartableEndpoint(SettingsHolder settings, IBuilder builder, FeatureActiv
 
             await transportInfrastructure.Start().ConfigureAwait(false);
 
-            var featureRunner = await StartFeatures(messageSession).ConfigureAwait(false);
-
             AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
 
-            mainPipeline = new Pipeline<ITransportReceiveContext>(builder, settings, pipelineConfiguration.Modifications);
+            var mainPipeline = new Pipeline<ITransportReceiveContext>(builder, settings, pipelineConfiguration.Modifications);
```

Comment:
I think we need to reference the main pipeline in the startable endpoint. It could get GCed if we don't